controllers/capabilities/mapping_rules.go

```diff
@@ -105,9 +114,12 @@ func (t *ProductThreescaleReconciler) syncMappingRules(_ interface{}) error {
 
 func (t *ProductThreescaleReconciler) processNotDesiredMappingRules(notDesiredList []threescaleapi.MappingRuleItem) error {
 	for _, mappingRule := range notDesiredList {
-		err := t.productEntity.DeleteMappingRule(mappingRule.ID)
-		if err != nil {
-			return err
+		if mappingRule.HTTPMethod == "" || mappingRule.Pattern == "" || mappingRule.Pattern == "/" {
```

Comment:
notDesiredMapping rule is not just a rule without http method, pattern or pattern = /
It's also rules that exist in 3scale but do not exist on the CR. Please revert this change.